src/libs/actions/OnyxUpdates.ts

```diff
@@ -34,11 +34,14 @@ function applyHTTPSOnyxUpdates(request: Request, response: Response) {
     return onyxDataUpdatePromise
         .then(() => {
             // Handle the request's success/failure data (client-side data)
-            if (response.jsonCode === 200 && request.successData) {
-                return updateHandler(request.successData);
+            const successData = request.successData?.length ? request.successData : request.finallyData;
```

Comment:
Hmmm ok I see what you did here, but I think what we want is to more closely emulate the behavior of native JS promises. TL;DR – this will apply `finallyData` only if `successData` (or `failureData` in the event of `response.jsonCode !== 200`) are absent.

You probably already know this, but try running this code in a Node.js REPL:

```node
➜  node
Welcome to Node.js v20.9.0.
Type ".help" for more information.
> function myAsyncFunc(shouldReject = false) {
...     return new Promise((resolve, reject) =>
...         setTimeout(() => {
...             if (shouldReject) {
...                 reject();
...             } else {
...                 resolve();
...             }
...         }, 1000),
...     );
... }
undefined
> function run(shouldReject = false) {
...     myAsyncFunc(shouldReject)
...         .then(() => console.log('Success!'))
...         .catch(() => console.log('Failure!'))
...         .finally(() => console.log('Finally!'));
... }
undefined
> run()
undefined
> Success!
Finally!

> run(true)
undefined
> Failure!
Finally!
```

Notice that in the success case, _both_ `.then` and `.finally` are executed, and same for `.catch` and `.finally` in the failure case. So I think that's the behavior we should implement here, like this:

```diff
return onyxDataUpdatePromise
    .then(() => {
        // Handle the request's success/failure data (client-side data)
        if (response.jsonCode === 200 && request.successData) {
            return updateHandler(request.successData);
        }
        if (response.jsonCode !== 200 && request.failureData) {
            return updateHandler(request.failureData);
        }
        return Promise.resolve();
    })
+   .then(() => {
+       if (request.finallyData) {
+           return updateHandler(request.finallyData);
+       }
+       return Promise.resolve();
+   })
    .then(() => {
        console.debug('[OnyxUpdateManager] Done applying HTTPS update');
        return Promise.resolve(response);
    });
```

Reply:
🤔 I'm not totally sure I follow... For the update you suggested, if only `finallyData` is included we'd resolve the promise and then go on to update the handler, which seems incorrect? 

Also, perhaps `finallyData` is the wrong name for this variable, as it implies data being returned _after_ success/failure and `finally` is already key term used in JS. Maybe `responseData` is a better alternative?

Reply:
> For the update you suggested, if only finallyData is included we'd resolve the promise and then go on to update the handler, which seems incorrect?

That seems correct to me – you can provide `finallyData` instead of `successData` or `failureData`, and in that case the updateHandler _should_ be called to perform the Onyx updates contained in `finallyData`

Reply:
Chatted about this a lil' more and came to an agreement on this!